stress: failed test in cockroach/server/server.test: TestNodeJoin

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/0958303ee7be3dd3fe7281c7008f520c53c8ebbd

Stress build found a failed test:

```
=== RUN   TestNodeJoin
I160418 04:54:17.736087 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160418 04:54:17.738845 storage/replica_command.go:1446  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405820h54m18.738504155s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-18 04:54:17.738739679 +0000 UTC]
W160418 04:54:17.748753 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
W160418 04:54:17.748806 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
I160418 04:54:17.748986 gossip/gossip.go:914  starting client to 127.0.0.1:42530
I160418 04:54:17.751367 server/node.go:373  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:8435806208 RangeCount:0}
I160418 04:54:17.751408 server/node.go:285  node ID 1 initialized
I160418 04:54:17.751480 storage/stores.go:287  read 0 node addresses from persistent storage
I160418 04:54:17.751566 server/node.go:494  connecting to gossip network to verify cluster ID...
I160418 04:54:17.757626 server/node.go:515  node connected via gossip and verified as part of cluster {"22e01827-c6e9-497c-9ee6-ac27b43de45e"}
I160418 04:54:17.757652 server/node.go:338  [node=1] Started node with [[]=] engine(s) and attributes []
I160418 04:54:17.757665 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160418 04:54:17.757960 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
W160418 04:54:17.758339 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
W160418 04:54:17.758378 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
I160418 04:54:17.758518 gossip/gossip.go:914  starting client to 127.0.0.1:42530
I160418 04:54:17.760416 server/node.go:360  store store=0:0 ([]=) not bootstrapped
I160418 04:54:17.760444 storage/stores.go:287  read 0 node addresses from persistent storage
I160418 04:54:17.760493 server/node.go:494  connecting to gossip network to verify cluster ID...
I160418 04:54:17.773608 gossip/client.go:83  closing client to node 1 (127.0.0.1:42530): gossip/client.go:177: stopping outgoing client to node 1 (127.0.0.1:42530); loopback connection
I160418 04:54:17.773939 server/node.go:515  node connected via gossip and verified as part of cluster {"22e01827-c6e9-497c-9ee6-ac27b43de45e"}
I160418 04:54:17.774028 storage/stores.go:302  wrote 1 node addresses to persistent storage
I160418 04:54:17.778381 kv/dist_sender.go:313  unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I160418 04:54:17.779037 server/node.go:276  new node allocated ID 2
I160418 04:54:17.779120 server/node.go:338  [node=2] Started node with [[]=] engine(s) and attributes []
I160418 04:54:17.779262 storage/replica_command.go:1900  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160418 04:54:17.779394 storage/stores.go:302  wrote 1 node addresses to persistent storage
I160418 04:54:17.780368 server/node.go:477  bootstrapped store store=2:2 ([]=)
I160418 04:54:17.781029 stopper.go:352  draining; tasks left:
1      server/node.go:419
I160418 04:54:17.781145 stopper.go:352  draining; tasks left:
1      storage/queue.go:383
1      server/node.go:741
1      kv/txn_coord_sender.go:893
I160418 04:54:17.781211 stopper.go:352  draining; tasks left:
1      storage/queue.go:383
1      server/node.go:741
E160418 04:54:17.781410 storage/queue.go:446  [split] range=1 [/Min-/Max): error: storage/split_queue.go:103: unable to split range=1 [/Min-/Max) at key "/Table/11/0": failed to send RPC: too many errors encountered (1 of 1 total): rpc error: code = 13 desc = transport is closing
I160418 04:54:17.781453 stopper.go:352  draining; tasks left:
1      server/node.go:741
I160418 04:54:18.781463 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42530: getsockopt: connection refused"; Reconnecting to "127.0.0.1:42530"
I160418 04:54:20.310245 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42530: getsockopt: connection refused"; Reconnecting to "127.0.0.1:42530"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x38 pc=0xb95052]

goroutine 2239 [running]:
panic(0x15f9fe0, 0xc82000e0e0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/storage.(*Store).Send.func1(0xc8213bb258, 0xc8213bb2c8, 0x14465b1104a08458, 0x0, 0xc8213bb2c0, 0xc820752000)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1558 +0x6e
panic(0x15f9fe0, 0xc82000e0e0)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
github.com/cockroachdb/cockroach/storage.(*treeContext).insertCase4(0xc8205232c0, 0xc820710460, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:395 +0xe2
github.com/cockroachdb/cockroach/storage.(*treeContext).insertCase3(0xc8205232c0, 0xc820710460, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:378 +0x3c5
github.com/cockroachdb/cockroach/storage.(*treeContext).insertCase2(0xc8205232c0, 0xc820710460, 0x1)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:351 +0xa1
github.com/cockroachdb/cockroach/storage.(*treeContext).insertCase1(0xc8205232c0, 0xc820710460, 0xc8213b9e18)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:338 +0xee
github.com/cockroachdb/cockroach/storage.(*treeContext).insert(0xc8205232c0, 0xc820710460, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:328 +0xba
github.com/cockroachdb/cockroach/storage.InsertRange(0xc8201c22c0, 0xc82063e000, 0xc8204e5118, 0x1, 0x8, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/range_tree.go:282 +0xb2
github.com/cockroachdb/cockroach/storage.(*Replica).AdminSplit.func1(0xc8201c22c0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:1931 +0x485
github.com/cockroachdb/cockroach/client.(*DB).Txn.func1(0xc8201c22c0, 0xc820970560, 0x4)
	/go/src/github.com/cockroachdb/cockroach/client/db.go:438 +0x25
github.com/cockroachdb/cockroach/client.(*Txn).Exec(0xc8201c22c0, 0x1d80101, 0x0, 0x0, 0xc8213ba388, 0x0)
	/go/src/github.com/cockroachdb/cockroach/client/txn.go:531 +0x40b
github.com/cockroachdb/cockroach/client.(*DB).Txn(0xc82096e320, 0xc8213ba688, 0xc8208ae678)
	/go/src/github.com/cockroachdb/cockroach/client/db.go:436 +0x261
github.com/cockroachdb/cockroach/storage.(*Replica).AdminSplit(0xc8201300e0, 0x7fa26d35cf08, 0xc820475500, 0xc8204e5110, 0x2, 0x8, 0x0, 0x0, 0x0, 0xc8204e5118, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica_command.go:1902 +0xe1a
github.com/cockroachdb/cockroach/storage.(*Replica).addAdminCmd(0xc8201300e0, 0x7fa26d35cf08, 0xc820475500, 0x14465b1104a080cb, 0x0, 0x100000001, 0x1, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1036 +0x876
github.com/cockroachdb/cockroach/storage.(*Replica).Send(0xc8201300e0, 0x7fa26d35cf08, 0xc820475500, 0x14465b1104a080cb, 0x0, 0x100000001, 0x1, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:782 +0x20c
github.com/cockroachdb/cockroach/storage.(*Store).Send(0xc820752000, 0x7fa26d35cf08, 0xc820475500, 0x14465b1104a080cb, 0x0, 0x100000001, 0x1, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1640 +0xd71
github.com/cockroachdb/cockroach/storage.(*Stores).Send(0xc8205ee740, 0x7fa26d35cf08, 0xc820474d20, 0x0, 0x0, 0x100000001, 0x1, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/storage/stores.go:178 +0x501
github.com/cockroachdb/cockroach/server.(*Node).Batch.func2()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:729 +0x55b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask(0xc820482000, 0xc8213bb870, 0x0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:166 +0xd9
github.com/cockroachdb/cockroach/server.(*Node).Batch(0xc820844900, 0x7fa26d35cf08, 0xc820474060, 0xc8204823f0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/server/node.go:741 +0x2c0
github.com/cockroachdb/cockroach/roachpb._Internal_Batch_Handler(0x172d980, 0xc820844900, 0x7fa26d35cf08, 0xc820474060, 0xc820967800, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1306 +0x132
google.golang.org/grpc.(*Server).processUnaryRPC(0xc8205a6000, 0x7fa26d395d78, 0xc820834120, 0xc82016c8c0, 0xc820f1c200, 0x21e2760, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/server.go:497 +0xe13
google.golang.org/grpc.(*Server).handleStream(0xc8205a6000, 0x7fa26d395d78, 0xc820834120, 0xc82016c8c0, 0x0)
	/go/src/google.golang.org/grpc/server.go:646 +0x109d
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc82081e140, 0xc8205a6000, 0x7fa26d395d78, 0xc820834120, 0xc82016c8c0)
	/go/src/google.golang.org/grpc/server.go:323 +0xa0
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/google.golang.org/grpc/server.go:324 +0x9a

goroutine 1 [chan receive]:
testing.RunTests(0x1abad28, 0x21ec620, 0x2e, 0x2e, 0xc8202c5801)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc8204c2f08, 0x5d6463)
	/usr/local/go/src/testing/testing.go:515 +0x81
main.main()
	github.com/cockroachdb/cockroach/server/_test/_testmain.go:146 +0x117

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 52 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x24c5f20)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:994 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:595 +0x8a

goroutine 2367 [runnable]:
sync.runtime_Syncsemacquire(0xc8213ee050)
	/usr/local/go/src/runtime/sema.go:241 +0x201
sync.(*Cond).Wait(0xc8213ee040)
	/usr/local/go/src/sync/cond.go:63 +0x9b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Quiesce(0xc820482000)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:354 +0x1eb
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Stop(0xc820482000)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:300 +0x1a7
github.com/cockroachdb/cockroach/server.TestNodeJoin(0xc820728000)
	/go/src/github.com/cockroachdb/cockroach/server/node_test.go:261 +0x815
testing.tRunner(0xc820728000, 0x21ec830)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 2240 [select]:
github.com/cockroachdb/cockroach/storage.(*idAllocator).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/id_alloc.go:131 +0x910
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8204e5230)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2378 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a6000)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2516 [select]:
github.com/cockroachdb/cockroach/rpc.(*Context).runHeartbeat(0xc8204820e0, 0xc820f20280, 0xc82081e710, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:209 +0x5e5
github.com/cockroachdb/cockroach/rpc.(*Context).GRPCDial.func1()
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:157 +0x66
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820eff6e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2422 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:794 +0x13d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc821380b40)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2431 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startComputePeriodicMetrics.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:561 +0x23b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820516460)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2548 [IO wait]:
net.runtime_pollWait(0x7fa26d2e4c80, 0x72, 0xc82006b400)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820762760, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820762760, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820762700, 0xc82006b400, 0x400, 0x400, 0x0, 0x7fa26d350028, 0xc82007c080)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820804060, 0xc82006b400, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc820620450, 0x7fa26d391ba0, 0xc820804060, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:460 +0xcc
crypto/tls.(*Conn).readRecord(0xc8208c4300, 0x1abb617, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:562 +0x2d1
crypto/tls.(*Conn).Read(0xc8208c4300, 0xc821382000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:939 +0x167
bufio.(*Reader).fill(0xc820443e60)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc820443e60, 0xc8204d4cf8, 0x9, 0x9, 0xc81ffee1c8, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0x7fa26d395d28, 0xc820443e60, 0xc8204d4cf8, 0x9, 0x9, 0x9, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:297 +0xe6
io.ReadFull(0x7fa26d395d28, 0xc820443e60, 0xc8204d4cf8, 0x9, 0x9, 0xc82033e6f8, 0x0, 0x0)
	/usr/local/go/src/io/io.go:315 +0x62
golang.org/x/net/http2.readFrameHeader(0xc8204d4cf8, 0x9, 0x9, 0x7fa26d395d28, 0xc820443e60, 0x0, 0xc800000000, 0x0, 0x0)
	/go/src/golang.org/x/net/http2/frame.go:236 +0xa5
golang.org/x/net/http2.(*Framer).ReadFrame(0xc8204d4cc0, 0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/net/http2/frame.go:463 +0x106
google.golang.org/grpc/transport.(*framer).readFrame(0xc82033e540, 0x0, 0x0, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/http_util.go:406 +0x3d
google.golang.org/grpc/transport.(*http2Client).reader(0xc82049b680)
	/go/src/google.golang.org/grpc/transport/http2_client.go:757 +0x109
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:200 +0x159a

goroutine 2377 [runnable]:
sync.runtime_Semacquire(0xc82081e14c)
	/usr/local/go/src/runtime/sema.go:47 +0x26
sync.(*WaitGroup).Wait(0xc82081e140)
	/usr/local/go/src/sync/waitgroup.go:127 +0xb4
google.golang.org/grpc.(*Server).serveStreams(0xc8205a6000, 0x7fa26d395d78, 0xc820834120)
	/go/src/google.golang.org/grpc/server.go:326 +0x167
google.golang.org/grpc.(*Server).serveNewHTTP2Transport(0xc8205a6000, 0x7fa26d395c50, 0xc8208e6000, 0x7fa26d395cb0, 0xc82043a2c0)
	/go/src/google.golang.org/grpc/server.go:312 +0x49d
google.golang.org/grpc.(*Server).handleRawConn(0xc8205a6000, 0x7fa26d2de800, 0xc820160018)
	/go/src/google.golang.org/grpc/server.go:289 +0x4ee
created by google.golang.org/grpc.(*Server).Serve
	/go/src/google.golang.org/grpc/server.go:261 +0x372

goroutine 2423 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:865 +0x3cf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc821380b60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2383 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a60a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2379 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a6020)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2381 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a6060)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2444 [select]:
github.com/cockroachdb/cockroach/rpc.(*Context).runHeartbeat(0xc8204820e0, 0xc820758640, 0xc8201083f0, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:209 +0x5e5
github.com/cockroachdb/cockroach/rpc.(*Context).GRPCDial.func1()
	/go/src/github.com/cockroachdb/cockroach/rpc/context.go:157 +0x66
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820499a70)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2273 [select]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).bootstrap.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:758 +0x219
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc821380320)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2368 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc82037e240)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:164 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:165 +0x8f4

goroutine 2418 [select]:
github.com/cockroachdb/cockroach/gossip.(*Gossip).manage.func1()
	/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:787 +0x33b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc821380340)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2233 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820482a80, 0xecea65f79, 0x2d2a6571, 0x24c5600, 0xc8213ee080, 0xc820482000, 0xc8201300e0, 0xc820616698)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:162 +0x568
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8201300e0, 0xc820498e40)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:204 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc8204990e0, 0xc820485200)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:237 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:206 +0x21e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820971de0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2382 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a6080)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2547 [select]:
google.golang.org/grpc/transport.(*http2Client).controller(0xc82049b680)
	/go/src/google.golang.org/grpc/transport/http2_client.go:835 +0x5da
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:194 +0x153b

goroutine 2419 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc820852ea0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:239 +0xd21
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:224 +0x20
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820108800)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2433 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc82036c2a0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:164 +0x3a
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:165 +0x8f4

goroutine 2384 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8204827e0, 0xecea65f79, 0x2d2af87e, 0x24c5600, 0xc8213ee080, 0xc820482000, 0xc8201300e0, 0xc820615698)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:162 +0x568
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8201300e0, 0xc820498e40)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:204 +0x78
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc820498f30, 0xc8207f6040)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:237 +0x246
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:206 +0x21e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a60c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2515 [select]:
google.golang.org/grpc.(*Conn).transportMonitor(0xc82027c1c0)
	/go/src/google.golang.org/grpc/clientconn.go:510 +0x1d3
google.golang.org/grpc.NewConn.func1(0xc82027c1c0)
	/go/src/google.golang.org/grpc/clientconn.go:321 +0x1b5
created by google.golang.org/grpc.NewConn
	/go/src/google.golang.org/grpc/clientconn.go:322 +0x4dd

goroutine 2232 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8209710c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2380 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:356 +0x303
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc8200a6040)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2424 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:884 +0x3cf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc821380b80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2432 [select]:
github.com/cockroachdb/cockroach/server.(*Node).startGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/server/node.go:528 +0x3e3
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc820516480)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62

goroutine 2421 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1878 +0xbee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820482000, 0xc82081e470)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62


ERROR: exit status 2
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
10 runs so far, 0 failures, over 15s
16 runs so far, 0 failures, over 20s
17 runs so far, 0 failures, over 25s
28 runs so far, 0 failures, over 30s
32 runs so far, 0 failures, over 35s
36 runs so far, 0 failures, over 40s
46 runs so far, 0 failures, over 45s
49 runs so far, 0 failures, over 50s
56 runs so far, 0 failures, over 55s
64 runs so far, 0 failures, over 1m0s
66 runs so far, 0 failures, over 1m5s
78 runs so far, 0 failures, over 1m10s
81 runs so far, 0 failures, over 1m15s
89 runs so far, 0 failures, over 1m20s
96 runs so far, 0 failures, over 1m25s
101 runs so far, 0 failures, over 1m30s
109 runs so far, 0 failures, over 1m35s
113 runs so far, 0 failures, over 1m40s
121 runs so far, 0 failures, over 1m45s
128 runs so far, 0 failures, over 1m50s
132 runs so far, 0 failures, over 1m55s
138 runs so far, 0 failures, over 2m0s
145 runs so far, 0 failures, over 2m5s
149 runs so far, 0 failures, over 2m10s
158 runs so far, 0 failures, over 2m15s
162 runs so far, 0 failures, over 2m20s
168 runs so far, 0 failures, over 2m25s
174 runs so far, 0 failures, over 2m30s
179 runs so far, 0 failures, over 2m35s
188 runs so far, 0 failures, over 2m40s
191 runs so far, 0 failures, over 2m45s
198 runs so far, 0 failures, over 2m50s
206 runs so far, 0 failures, over 2m55s
209 runs so far, 0 failures, over 3m0s
218 runs so far, 0 failures, over 3m5s
224 runs so far, 0 failures, over 3m10s
227 runs so far, 0 failures, over 3m15s
238 runs so far, 0 failures, over 3m20s
241 runs so far, 0 failures, over 3m25s
246 runs so far, 0 failures, over 3m30s
257 runs so far, 0 failures, over 3m35s
260 runs so far, 0 failures, over 3m40s
267 runs so far, 0 failures, over 3m45s
273 runs so far, 0 failures, over 3m50s
278 runs so far, 0 failures, over 3m55s
285 runs so far, 0 failures, over 4m0s
289 runs so far, 0 failures, over 4m5s
296 runs so far, 0 failures, over 4m10s
304 runs so far, 0 failures, over 4m15s
311 runs so far, 0 failures, over 4m20s
317 runs so far, 0 failures, over 4m25s
322 runs so far, 0 failures, over 4m30s
328 runs so far, 0 failures, over 4m35s
335 runs so far, 0 failures, over 4m40s
341 runs so far, 0 failures, over 4m45s
346 runs so far, 0 failures, over 4m50s
353 runs so far, 0 failures, over 4m55s
360 runs so far, 0 failures, over 5m0s
366 runs so far, 0 failures, over 5m5s
371 runs so far, 0 failures, over 5m10s
379 runs so far, 0 failures, over 5m15s
382 runs so far, 0 failures, over 5m20s
390 runs so far, 0 failures, over 5m25s
397 runs so far, 0 failures, over 5m30s
401 runs so far, 0 failures, over 5m35s
408 runs so far, 0 failures, over 5m40s
414 runs so far, 0 failures, over 5m45s
420 runs so far, 0 failures, over 5m50s
425 runs so far, 0 failures, over 5m55s
433 runs so far, 0 failures, over 6m0s
439 runs so far, 0 failures, over 6m5s
443 runs so far, 0 failures, over 6m10s
453 runs so far, 0 failures, over 6m15s
454 runs so far, 0 failures, over 6m20s
464 runs so far, 0 failures, over 6m25s
469 runs so far, 0 failures, over 6m30s
476 runs so far, 0 failures, over 6m35s
482 runs so far, 0 failures, over 6m40s
486 runs so far, 0 failures, over 6m45s
495 runs so far, 0 failures, over 6m50s
500 runs so far, 0 failures, over 6m55s
505 runs so far, 0 failures, over 7m0s
515 runs so far, 0 failures, over 7m5s
517 runs so far, 0 failures, over 7m10s
528 runs so far, 0 failures, over 7m15s
532 runs so far, 0 failures, over 7m20s
533 runs so far, 0 failures, over 7m25s
547 runs so far, 0 failures, over 7m30s
548 runs so far, 0 failures, over 7m35s
556 runs so far, 0 failures, over 7m40s
562 runs so far, 0 failures, over 7m45s
566 runs so far, 0 failures, over 7m50s
575 runs so far, 0 failures, over 7m55s
580 runs so far, 0 failures, over 8m0s
584 runs so far, 0 failures, over 8m5s
593 runs so far, 0 failures, over 8m10s
598 runs so far, 0 failures, over 8m15s
603 runs so far, 0 failures, over 8m20s
609 runs so far, 0 failures, over 8m25s
616 runs so far, 0 failures, over 8m30s
622 runs so far, 0 failures, over 8m35s
627 runs so far, 0 failures, over 8m40s
634 runs so far, 0 failures, over 8m45s
642 runs so far, 0 failures, over 8m50s
647 runs so far, 0 failures, over 8m55s
654 runs so far, 0 failures, over 9m0s
658 runs so far, 0 failures, over 9m5s
666 runs so far, 0 failures, over 9m10s
672 runs so far, 0 failures, over 9m15s
676 runs so far, 0 failures, over 9m20s
684 runs so far, 0 failures, over 9m25s
689 runs so far, 0 failures, over 9m30s
696 runs so far, 0 failures, over 9m35s
702 runs so far, 0 failures, over 9m40s
707 runs so far, 0 failures, over 9m45s
715 runs so far, 0 failures, over 9m50s
720 runs so far, 0 failures, over 9m55s
726 runs so far, 0 failures, over 10m0s
733 runs so far, 0 failures, over 10m5s
739 runs so far, 0 failures, over 10m10s
747 runs so far, 0 failures, over 10m15s
752 runs so far, 0 failures, over 10m20s
759 runs so far, 0 failures, over 10m25s
766 runs so far, 0 failures, over 10m30s
769 runs so far, 0 failures, over 10m35s
777 runs so far, 0 failures, over 10m40s
784 runs so far, 0 failures, over 10m45s
791 runs so far, 0 failures, over 10m50s
795 runs so far, 0 failures, over 10m55s
802 runs so far, 0 failures, over 11m0s
810 runs so far, 0 failures, over 11m5s
815 runs so far, 0 failures, over 11m10s
819 runs so far, 0 failures, over 11m15s
827 runs so far, 0 failures, over 11m20s
834 runs so far, 0 failures, over 11m25s
836 runs so far, 0 failures, over 11m30s
845 runs so far, 0 failures, over 11m35s
851 runs so far, 0 failures, over 11m40s
854 runs completed, 1 failures, over 11m42s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/kv/txn_coord_sender.go
+++ b/kv/txn_coord_sender.go
@@ -829,7 +829,7 @@ func (tc *TxnCoordSender) updateState(
 	default:
 		if pErr.GetTxn() != nil {
 			if pErr.CanRetry() {
-				panic("Retryable internal error must not happen at this level")
+				log.Fatalf("retryable internal error must not happen at this level: %s", pErr)
 			} else {
 				// Do not clean up the transaction here since the client might still
 				// want to continue the transaction. For example, a client might

--- a/server/node_test.go
+++ b/server/node_test.go
@@ -109,6 +109,9 @@ func createAndStartTestNode(addr net.Addr, engines []engine.Engine, gossipBS net
 	if err := node.start(addr, engines, roachpb.Attributes{}); err != nil {
 		t.Fatal(err)
 	}
+	if err := waitForInitialSplits(node.ctx.DB); err != nil {
+		t.Fatal(err)
+	}
 	return grpcServer, addr, node, stopper
 }
 

--- a/server/testserver.go
+++ b/server/testserver.go
@@ -274,10 +274,14 @@ func ExpectedInitialRangeCount() int {
 // splits at startup. If the expected range count is not reached within a
 // configured timeout, an error is returned.
 func (ts *TestServer) WaitForInitialSplits() error {
+	return waitForInitialSplits(ts.DB())
+}
+
+func waitForInitialSplits(db *client.DB) error {
 	expectedRanges := ExpectedInitialRangeCount()
 	return util.RetryForDuration(initialSplitsTimeout, func() error {
 		// Scan all keys in the Meta2Prefix; we only need a count.
-		rows, pErr := ts.DB().Scan(keys.Meta2Prefix, keys.MetaMax, 0)
+		rows, pErr := db.Scan(keys.Meta2Prefix, keys.MetaMax, 0)
 		if pErr != nil {
 			return pErr.GoError()
 		}

